Engine/EditorUI/Panel/PanelProjectExplorer.cpp

```diff
@@ -81,7 +74,14 @@ void PanelProjectExplorer::Render()
 		{
 			ImGui::BeginChild("Project File Explorer Drop Target");
 			hovered = ImGui::IsWindowHovered() ? true : hovered;
-			ShowFilesInExplorer();
+			if (searching_file == "")
+			{
+				ShowFilesInExplorer();
+			}
+			else
+			{
+				SearchFilesInExplorer(App->filesystem->assets_folder_path);
```

Comment:
Searching for files every frame could be inefficient. Instead of doing here, it's better to look for them in line 49. `ImGui::InputText()` returns true when the string is changed